### PR TITLE
HIPP-584: Refactor return types in database layer in api-hub-applications

### PR DESCRIPTION
**Pull request by PaulCDurham**
https://jira.tools.tax.service.gov.uk/browse/HIPP-584

Also applied ExceptionRaising to IdmsConnectorImpl to standardise exception behaviour

### CODE REVIEW

File: app/uk/gov/hmrc/apihubapplications/repositories/ApplicationsRepository.scala [hunk 1]
```diff
@@ -85,12 +91,19 @@ class ApplicationsRepository @Inject()
             options     = ReplaceOptions().upsert(false)
           )
           .toFuture()
-          .map(_.getModifiedCount == 1)
-      case None => Future.successful(false)
+          .map(
+            result =>
+              if (result.getModifiedCount > 0) {
```
Comment (laurahk) [inline app/uk/gov/hmrc/apihubapplications/repositories/ApplicationsRepository.scala:96]: The logic here is slightly changed. If it was greater than 0 e.g. 2, should that be a success or an error? Presumably we only expect a single application to be updated?

Comment (PaulCDurham) [reply app/uk/gov/hmrc/apihubapplications/repositories/ApplicationsRepository.scala:96]: Given the unique index on _id that can't actually happen

Comment (laurahk) [reply app/uk/gov/hmrc/apihubapplications/repositories/ApplicationsRepository.scala:96]: Ah ok cool 👍

File: app/uk/gov/hmrc/apihubapplications/services/ApplicationsService.scala [hunk 2]
```diff
@@ -105,26 +105,25 @@ class ApplicationsService @Inject()(
...
           repositoryUpdated <- repositoryUpdate
           idmsUpdated <- idmsUpdate
         } yield (repositoryUpdated, idmsUpdated) match {
-          case (true, Right(_)) => Right(true)
+          case (Right(()), Right(_)) => Right(true)
           case (_, Left(e)) => Left(e)
           case _ => Right(false)
         }
-      case None => Future.successful(Right(false))
+      case _ => Future.successful(Right(false))
```
[the remark this answers is not in the document]
Comment (laurahk) [reply app/uk/gov/hmrc/apihubapplications/services/ApplicationsService.scala:120]: Possibly even the `findById` function in ApplicationsService should return it too?

**Comment by laurahk** [comment]
Loving the refactor - code is so much clearer!

**Review by laurahk — APPROVED** [review]

File: app/uk/gov/hmrc/apihubapplications/services/ApplicationsService.scala (hunk 2 above)
[the remark this answers is not in the document]
Comment (PaulCDurham) [reply app/uk/gov/hmrc/apihubapplications/services/ApplicationsService.scala:120]: I've left most of the service layer to ticket HIPP-583. I've picked that one up now and will try and push the Either[etc] pattern up to the controller layer which should make a few bits of code like this neater. I didn't want too much chaos on one ticket.

**Pull request merged**